validator/client/validator.go

```diff
@@ -931,6 +935,99 @@ func (v *validator) logDuties(slot types.Slot, duties []*ethpb.DutiesResponse_Du
 	}
 }
 
+// SetPubKeyToValidatorIndexMap sets the pubKeyToValidatorIndexMap which caches the mapping of public keys to validator indices.
+func (v *validator) SetPubKeyToValidatorIndexMap(ctx context.Context, km keymanager.IKeymanager) error {
```

Comment:
Maybe we can find a way to not export this or couple it where it is needed. It seems like this is only used during the beacon proposer preparation, so ideally we can move it within that method

Reply:
removed this function